types/next-auth/client.d.ts

```diff
@@ -59,6 +63,7 @@ declare function signin(
     authorizationParams?: string | string[][] | Record<string, unknown> | URLSearchParams
 ): Promise<void>;
 declare const signIn: typeof signin;
+declare function signout(data: { callbackUrl?: string, redirect: false }): Promise<SignOutResponse>;
```

Comment:
I think to be more faithful with [what's in the source code](https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js#L285-L291) we should change this to 💭
```ts
declare function signout(options: { callbackUrl?: string, redirect: true }): Promise<void>;
declare function signout(options?: { callbackUrl?: string, redirect?: boolean }): Promise<SignOutResponse>;
```
Same for the `signIn` declaration overload above:
```ts
declare function signin(
    provider: 'credentials' | 'email',
    data: Record<string, unknown> & {
        callbackUrl?: string;
        redirect: true;
    },
    authorizationParams?: string | string[][] | Record<string, unknown> | URLSearchParams
  ): Promise<void>;
declare function signin(
    provider?: string,
    data?: Record<string, unknown> & {
        callbackUrl?: string;
        redirect?: boolean;
    },
    authorizationParams?: string | string[][] | Record<string, unknown> | URLSearchParams
): Promise<SignInResponse>;
```
What do you think?

Reply:
@lluia You are right, but with one modification I think.
Since the [default value of redirect is true](https://next-auth.js.org/getting-started/client#using-the-redirect-false-option), shoudn't it be something like this?

```diff
-	declare function signout(options: { callbackUrl?: string, redirect: true }): Promise<void>;
-	declare function signout(options?: { callbackUrl?: string, redirect?: boolean }): Promise<SignOutResponse>;

+	declare function signout(options: { callbackUrl?: string, redirect?: true }): Promise<void>;
+	declare function signout(options?: { callbackUrl?: string, redirect: boolean }): Promise<SignOutResponse>;
```

```diff
	declare function signin(
	    provider: 'credentials' | 'email',
	    data: Record<string, unknown> & {
	        callbackUrl?: string;
-	        redirect: true;
+	        redirect?: true;
	    },
	    authorizationParams?: string | string[][] | Record<string, unknown> | URLSearchParams
	  ): Promise<void>;
	declare function signin(
	    provider?: string,
	    data?: Record<string, unknown> & {
	        callbackUrl?: string;
-	        redirect?: boolean;
+	        redirect: boolean;
	    },
	    authorizationParams?: string | string[][] | Record<string, unknown> | URLSearchParams
	): Promise<SignInResponse>;
```

Reply:
Nice, I think your suggestion makes sense about making `redirect?` 👍🏽 . Sure thing with the interface, you can write it in a separate comment 💬